vhdbuilder/packer/backfill-cleanup.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash -x
+
+EXPIRATION_IN_HOURS=168
+# convert to seconds so we can compare it against the "tags.now" property in the resource group metadata
+(( expirationInSecs = ${EXPIRATION_IN_HOURS} * 60 * 60 ))
+# deadline = the "date +%s" representation of the oldest age we're willing to keep
+(( deadline=$(date +%s)-${expirationInSecs%.*} ))
+
+if [[ "${MODE}" != "windowsVhdMode" ]]; then
+  exit 0
+fi
+
+# we will clean old sig image-definitions/versions/galleries, RGs only if 
+# we have overriden the pipeline variable SKIP_IMAGE_DELETION_BACKFILL to False
+# or we are running on production pipelines
+CLEAN_OLD_WINDOWS_RESOURCES="False"
+if [[ "${SKIP_IMAGE_DELETION_BACKFILL}" == "False" || "${DRY_RUN}" == "False" ]]; then
```

Comment:
to clarify, we backfill resource deletion when this variable is overridden or during any build not in dry-run mode?

Reply:
should separate out chained conditionals with brackets:

[[ "${SKIP_IMAGE_DELETION}" == "False" ]] || [[ "${DRY_RUN}" == "False" ]]